cura/Machines/MachineNode.py

```diff
@@ -168,13 +168,17 @@ def preferredGlobalQuality(self) -> "QualityNode":
 
         return self.global_qualities.get(self.preferred_quality_type, next(iter(self.global_qualities.values())))
 
-    def isExcludedMaterial(self, material: MaterialNode) -> bool:
+    def isExcludedMaterialBaseFile(self, material_base_file: str) -> bool:
         """Returns whether the material should be excluded from the list of materials."""
         for exclude_material in self.exclude_materials:
-            if exclude_material in material["id"]:
+            if exclude_material in material_base_file:
                 return True
         return False
 
+    def isExcludedMaterial(self, material: MaterialNode) -> bool:
```

Comment:
Well, now we can set this function deprecated, as we are no more using it, right ?

Reply:
Done! :-)